src/main/java/ay2122s1_cs2103t_w16_2/btbb/logic/parser/order/FindOrderCommandParser.java

```diff
@@ -0,0 +1,65 @@
+package ay2122s1_cs2103t_w16_2.btbb.logic.parser.order;
+
+import static ay2122s1_cs2103t_w16_2.btbb.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static ay2122s1_cs2103t_w16_2.btbb.commons.util.CollectionUtil.requireAllNonNull;
+import static ay2122s1_cs2103t_w16_2.btbb.logic.parser.util.CliSyntax.PREFIX_CLIENT_ADDRESS;
+import static ay2122s1_cs2103t_w16_2.btbb.logic.parser.util.CliSyntax.PREFIX_CLIENT_EMAIL;
+import static ay2122s1_cs2103t_w16_2.btbb.logic.parser.util.CliSyntax.PREFIX_CLIENT_NAME;
+import static ay2122s1_cs2103t_w16_2.btbb.logic.parser.util.CliSyntax.PREFIX_CLIENT_PHONE;
+
+import java.util.List;
+import java.util.function.Function;
+
+import ay2122s1_cs2103t_w16_2.btbb.exception.ParseException;
+import ay2122s1_cs2103t_w16_2.btbb.logic.commands.order.FindOrderCommand;
+import ay2122s1_cs2103t_w16_2.btbb.logic.parser.Parser;
+import ay2122s1_cs2103t_w16_2.btbb.logic.parser.util.ArgumentMultimap;
+import ay2122s1_cs2103t_w16_2.btbb.logic.parser.util.ArgumentTokenizer;
+import ay2122s1_cs2103t_w16_2.btbb.logic.parser.util.Prefix;
+import ay2122s1_cs2103t_w16_2.btbb.model.order.Order;
+import ay2122s1_cs2103t_w16_2.btbb.model.shared.GenericStringPredicate;
+import ay2122s1_cs2103t_w16_2.btbb.model.shared.PredicateCollection;
+
+/**
+ * Parses input arguments and creates a new FindOrderCommand object.
+ */
+public class FindOrderCommandParser implements Parser<FindOrderCommand> {
+    private void addOrderPredicate(PredicateCollection<Order> predicateCollection, ArgumentMultimap argMultimap,
+                                   Prefix prefix, Function<Order, ?> getter) {
+        if (argMultimap.getValue(prefix).isEmpty()) {
+            return;
+        }
+
+        String trimmedSearchArg = argMultimap.getValue(prefix).get().trim();
+        if (trimmedSearchArg.isEmpty()) {
+            return;
+        }
+
+        List<String> keywords = List.of(trimmedSearchArg.split("\\s+"));
+        predicateCollection.addPredicate(new GenericStringPredicate<>(getter, keywords));
+    }
```

Comment:
This is an exact copy of addClientPredicate right? Can we change this whole thing? 

Firstly I am not a fan of how it mixes concerns and breaks the abstractions of the rest of the app. Usually the ParserUtil is in charge of ensuring inputs are valid. However with this function, we make inputs such as `find-o cn/alex ca/` valid, even though the param for `ca/` is empty.

In ParserUtil under `// Shared-level parsers:` can we add something like, or we can even create a class for keywords although I think it's not necessary.
```
    public static List<String> parseKeywords(String keywords) throws ParseException {
        requireNonNull(keywords);
        String trimmedKeywords = keywords.trim();
        if (trimmedKeywords.isEmpty()) {
            throw new ParseException(MESSAGE_INVALID_KEYWORD);
        }
        return List.of(trimmedKeywords.split("\\s+"));
    }
```

Secondly, this function is not extensible because it assumes that all `addXPredicates` will use `GenericStringPredicate`. This is not true when we search by quantity and date. 

Thirdly this function is an exact copy of `addClientPredicate` which should make us worried about DRY.

So after adding that parserUtil function, we can follow the rest of the app's convention in the `FindXParser`.
For example, use
```
predicateCollection.addPredicate(new GenericStringPredicate<>(Order::getClientName,
                        ParserUtil.parseKeywords(argMultimap.getValue(PREFIX_CLIENT_NAME).get())));
```